### PR TITLE
Adds 8.15 and bumps master to 8.16

### PR DESCRIPTION
**Pull request by KOTungseth**

### CODE REVIEW

**Comment by leemthompo** [comment]
@KOTungseth I reached out to team to cut `enterprise-search-js` 8.15 branch to fix CI

**Comment by leemthompo** [comment]
run docs-build

**Comment by KOTungseth** [comment]
run docs-build

**Comment by KOTungseth** [comment]
run docs-build

**Comment by KOTungseth** [comment]
run docs-build

**Comment by KOTungseth** [comment]
run docs-build

**Comment by colleenmcginnis** [comment]
@KOTungseth I think this needs to be merged before the test will pass: https://github.com/elastic/ingest-docs/pull/1162.

**Comment by georgewallace** [comment]
run docs-build

**Comment by KOTungseth** [comment]
run docs-build

File: shared/versions/stack/8.15.asciidoc [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+:esf_version:            master
+:ece-version-link:       current
+//////////
+Keep the :esf_version: attribute value as master.
+//////////
+
+//////////
+release-state can be: released | prerelease | unreleased
+//////////
+:release-state:          released
```
Comment (lcawl) [inline shared/versions/stack/8.15.asciidoc:26]:
```suggestion
:release-state:          unreleased
```

File: shared/versions/stack/8.15.asciidoc [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+//////////
+is-current-version can be: true | false
+//////////
+:is-current-version:    true
```
Comment (lcawl) [inline shared/versions/stack/8.15.asciidoc:31]:
```suggestion
:is-current-version:    false
```

**Review by lcawl — APPROVED** [review]
LGTM

**Pull request merged**